xdsl/interpreters/linalg.py

```diff
@@ -70,6 +70,7 @@ def run_add(
         assert isinstance(lhs, ShapedArray)
         assert isinstance(rhs, ShapedArray)
         assert isinstance(res, ShapedArray)
+        assert all(res.data_ptr[i] == 0 for i in range(len(res.data)))
```

Comment:
hmm, maybe the best would be to add a helper for this to the class directly, to make the code here a little simpler. Could you please have a go at this? There might be a method on `memoryview` that would give this directly, maybe something like number of set bits == 0

Reply:
If you feel like you'd rather do that later, please file an issue and we can do it some other time, this seems to work great already